save-agent/build.gradle.kts

```
@@ -123,3 +124,22 @@ kotlin.sourceSets.getByName("linuxX64Main").dependsOn(generatedKotlinSrc)
 tasks.withType<org.jetbrains.kotlin.gradle.dsl.KotlinCompile<*>>().configureEach {
     dependsOn(generateVersionFileTaskProvider)
 }
+
+/*
+ * On Windows, it's impossible to link a Linux executable against
+ * `io.ktor:ktor-client-curl` because `-lcurl` is not found by `ld`.
+ */
+private val linkDebugExecutableLinuxX64 by tasks
+linkDebugExecutableLinuxX64.onlyIf {
+    !DefaultNativePlatform.getCurrentOperatingSystem().isWindows
+}
```

I believe, using `tasks`'s delegate returns an instance of task eagerly, which contradicts recommended configuration avoidance in Gradle. The following will configure the task lazily, i.e. only if it's present on the current task graph.
```suggestion
tasks.named("linkDebugExecutableLinuxX64") {
    onlyIf {
        !DefaultNativePlatform.getCurrentOperatingSystem().isWindows
    }
}
```

---

Thanks!

Addressde with 90030226de0c77b4cf4636afb4fcbc8e7632d26d